From f693bd48a7be3ea957b6177b9b112d130dd63514 Mon Sep 17 00:00:00 2001
From: "shravanrn@gmail.com" <shravanrn@gmail.com>
Date: Mon, 11 Nov 2019 09:19:37 +0000
Subject: [PATCH 08/35] Bug 1572619 - Include RLBox Lucet integration repo to
 use wasm library sandboxes r=froydnj

Differential Revision: https://phabricator.services.mozilla.com/D43030

--HG--
extra : moz-landing-system : lando

diff --git a/Cargo.toml b/Cargo.toml
index ef0649707d95..2632958467cf 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,6 +61,7 @@ libudev-sys = { path = "dom/webauthn/libudev-sys" }
 serde_derive = { git = "https://github.com/servo/serde", branch = "deserialize_from_enums10" }
 winapi = { git = "https://github.com/froydnj/winapi-rs", branch = "aarch64" }
 packed_simd = { git = "https://github.com/hsivonen/packed_simd", branch = "rust_1_32" }
+rlbox_lucet_sandbox = { git = "https://github.com/PLSysSec/rlbox_lucet_sandbox/", rev="997c648eb0eaeaaa7a00a9eee20431f750b4e190" }
 nix = { git = "https://github.com/shravanrn/nix/", branch = "r0.13.1", rev="4af6c367603869a30fddb5ffb0aba2b9477ba92e" }
 
 [patch.crates-io.cranelift-codegen]
diff --git a/toolkit/library/gtest/rust/Cargo.toml b/toolkit/library/gtest/rust/Cargo.toml
index b2e9fcb8ba3a..d04e9f878115 100644
--- a/toolkit/library/gtest/rust/Cargo.toml
+++ b/toolkit/library/gtest/rust/Cargo.toml
@@ -25,6 +25,7 @@ gecko_profiler_parse_elf = ["gkrust-shared/gecko_profiler_parse_elf"]
 bitsdownload = ["gkrust-shared/bitsdownload"]
 new_xulstore = ["gkrust-shared/new_xulstore"]
 new_cert_storage = ["gkrust-shared/new_cert_storage"]
+wasm_library_sandboxing = ["gkrust-shared/wasm_library_sandboxing"]
 
 [dependencies]
 bench-collections-gtest = { path = "../../../../xpcom/rust/gtest/bench-collections" }
diff --git a/toolkit/library/rust/Cargo.toml b/toolkit/library/rust/Cargo.toml
index 1188696d17cc..907ed1ded50f 100644
--- a/toolkit/library/rust/Cargo.toml
+++ b/toolkit/library/rust/Cargo.toml
@@ -26,6 +26,7 @@ gecko_profiler_parse_elf = ["gkrust-shared/gecko_profiler_parse_elf"]
 bitsdownload = ["gkrust-shared/bitsdownload"]
 new_xulstore = ["gkrust-shared/new_xulstore"]
 new_cert_storage = ["gkrust-shared/new_cert_storage"]
+wasm_library_sandboxing = ["gkrust-shared/wasm_library_sandboxing"]
 
 [dependencies]
 gkrust-shared = { path = "shared" }
diff --git a/toolkit/library/rust/shared/Cargo.toml b/toolkit/library/rust/shared/Cargo.toml
index e6a28275c4fa..b7fe7e574a8d 100644
--- a/toolkit/library/rust/shared/Cargo.toml
+++ b/toolkit/library/rust/shared/Cargo.toml
@@ -38,6 +38,7 @@ cert_storage = { path = "../../../../security/manager/ssl/cert_storage", optiona
 bitsdownload = { path = "../../../components/bitsdownload", optional = true }
 storage = { path = "../../../../storage/rust" }
 bookmark_sync = { path = "../../../components/places/bookmark_sync", optional = true }
+rlbox_lucet_sandbox = { version = "0.1.0", optional = true }
 
 [build-dependencies]
 rustc_version = "0.2"
@@ -63,6 +64,7 @@ gecko_profiler = ["profiler_helper", "geckoservo/gecko_profiler"]
 gecko_profiler_parse_elf = ["profiler_helper/parse_elf"]
 new_xulstore = ["xulstore"]
 new_cert_storage = ["cert_storage"]
+wasm_library_sandboxing = ["rlbox_lucet_sandbox"]
 
 [lib]
 path = "lib.rs"
diff --git a/toolkit/library/rust/shared/lib.rs b/toolkit/library/rust/shared/lib.rs
index e5d74999a8f1..acd8c1368436 100644
--- a/toolkit/library/rust/shared/lib.rs
+++ b/toolkit/library/rust/shared/lib.rs
@@ -47,6 +47,9 @@ extern crate bookmark_sync;
 
 extern crate arrayvec;
 
+#[cfg(feature = "wasm_library_sandboxing")]
+extern crate rlbox_lucet_sandbox;
+
 use std::boxed::Box;
 use std::env;
 use std::ffi::{CStr, CString};
-- 
2.25.1

